packages/blocks/src/database-block/utils.ts

```diff
@@ -61,3 +70,80 @@ export const getTagColor = tagColorHelper();
 export function isDivider(action: DatabaseAction): action is Divider {
   return action.type === 'divider';
 }
+
+type InitLimitedLengthVEditorConfig = {
+  yText: Y.Text | string;
+  container: HTMLElement;
+  targetModel: BaseBlockModel;
+  maxLength: number;
+  readonly?: boolean;
+  handlers?: VirgoEventService<BaseTextAttributes>['_handlers'];
+  options?: VEditorOptions;
+};
+export function initLimitedLengthVEditor({
+  yText,
+  container,
+  targetModel,
+  maxLength,
+  readonly,
+  handlers = {},
+  options = {
+    defaultMode: 'rich',
+  },
+}: InitLimitedLengthVEditorConfig) {
+  const handlePaste = (event: ClipboardEvent) => {
+    const length = vEditor.yText.length;
+    const restLength = maxLength - length;
+    if (restLength <= 0) return;
+
+    const data = event.clipboardData?.getData('text/plain');
+    if (!data) return;
+
+    const vRange = vEditor.getVRange();
+    const text = data.length > restLength ? data.slice(0, restLength) : data;
+    if (vRange) {
+      vEditor.insertText(vRange, text);
+      vEditor.setVRange({
+        index: vRange.index + text.length,
+        length: 0,
+      });
+    }
+  };
+
+  const vEditor = new VEditor(yText, options);
+  setupVirgoScroll(targetModel.page, vEditor);
+  vEditor.mount(container);
+  vEditor.bindHandlers({
+    virgoInput: ctx => {
+      const length = vEditor.yText.length;
+      if (length >= maxLength && ctx.event.data) {
+        // prevent input
+        ctx.skipDefault = true;
+      }
+      return ctx;
+    },
+    paste: handlePaste,
+    virgoCompositionEnd: ctx => {
```

Comment:
@Flrande this sill looks a bit leaky since we still require handling yText on upper layer for adding length limit, let's land this for now, and hope for future improvements to further simplify this.

/cc @Saul-Mirone